org-static-blog.el

```diff
@@ -302,27 +302,36 @@ Preamble and Postamble are excluded, too."
        (search-backward "</div>")
        (point)))))
 
-(defun org-static-blog-get-url (post-filename)
-  "Generate a URL to the published POST-FILENAME."
-  (concat (org-static-blog-post-url
-           (org-static-blog-prepare-post-url post-filename)
-           (org-static-blog-get-date post-filename))
-          ".html"))
-
-(defun org-static-blog-prepare-post-url (post-filename)
-  "Cleanup and normalize POST-FILENAME to be sent to user-defined function"
-  (replace-regexp-in-string ".org$" ""
-   (replace-regexp-in-string
-    (concat "^\\("
-            (file-truename org-static-blog-posts-directory)
-            "\\|"
-            (file-truename org-static-blog-drafts-directory)
-            "\\)")
-    ""
-    post-filename)))
-
-(defun org-static-blog-post-url (post-filename post-datetime)
-  "Generate arbitrary post URL. This function is overridable by user"
+(defun org-static-blog-get-full-url (relative-url)
+  "Return absolute URL based on the RELATIVE-URL passed to the function."
+  (concat org-static-blog-publish-url relative-url))
+
+(defun org-static-blog-get-post-url (post-filename)
+  "Generate full URL to the published POST-FILENAME."
+  (org-static-blog-get-full-url
+          (org-static-blog-get-post-public-path post-filename)))
+
+(defun org-static-blog-get-post-public-path (post-filename)
+  "Return post filepath in public directory."
+  (org-static-blog-generate-post-path
+   (org-static-blog-get-relative-path post-filename)
+   (org-static-blog-get-date post-filename)))
+
+(defun org-static-blog-get-relative-path (post-filename)
+  "Remove absolute directory path from POST-FILENAME and return relative path to HTML file."
+  (replace-regexp-in-string ".org$" ".html"
+                            (replace-regexp-in-string
+                             (concat "^\\("
+                                     (file-truename org-static-blog-posts-directory)
+                                     "\\|"
+                                     (file-truename org-static-blog-drafts-directory)
+                                     "\\)")
+                             ""
+                             post-filename)))
+
+(defun org-static-blog-generate-post-path (post-filename post-datetime)
+  "Generate post public path (URL) based on POST-FILENAME and PORT-DATETIME.
```

Comment:
As far as I can tell, the other functions either end on `-path` or `-url`, depending on whether they return a local file path or a public url. This one seems to be the odd one out.